services/gam/src/main.rs

```diff
@@ -567,7 +567,7 @@ fn wrapped_main() -> ! {
                         // set before the leaving state due to interior-mutability issues (see the block under the debug statement
                         // titled "resolving visibility rules"). The fix is to simply not issue a context switch if it's the same app.
                         // It saves CPU thrashing and also works around this problem. (see issue #145)
-                        let authorized_switchers = vec![MAIN_MENU_NAME, ROOTKEY_MODAL_NAME, gam::STATUS_BAR_NAME];
+                        let authorized_switchers = [&[MAIN_MENU_NAME, ROOTKEY_MODAL_NAME, gam::STATUS_BAR_NAME], gam::EXPECTED_APP_CONTEXTS].concat().to_vec();
```

Comment:
@xobs pointed this line out to me, and yes...as the PR requestor has noted this opens a big security hole. 

One of the ideas inside Xous' UI layer is you have a static set of graphics contexts, so as to make it harder for someone to fabricate a UI that looks like another app to fool you into typing credentials into a fake app.

However, if you're going to allow arbitrary executable code on the device, creating an extra UI context is (a) necessary and (b) probably a lesser problem than the arbitrary executable code itself.

To that end, I think this would definitely need to be feature-gated -- this is not the sort of thing you'd want to have in the releases, it's a dev-only convenience.